records/2021-09-20.scalibility.notify-short-code.md

```diff
@@ -0,0 +1,50 @@
+
+## Additional considerations
+
+1. Short codes are supposed to be tied to a single service. As can be seen from our application(and subsequent rejection) to AWS last year for short codes (https://docs.google.com/document/d/1IbBHE_hDtXmV03FJFeB5wRWUekzmEP4Ky2x0a_lDBtg/edit), each short code is meant to be single purpose.
+2. Notify is a software service that is run by other services. The unsubscribe feature is managed by AWS (If a customer responds to the short code with 'STOP', AWS puts the user on a do not send list). AWS allows us to remove a single user from the unsubscribe list once every 30 days.
+Eg: Immigration, Refugee and Citizenship Canada (IRCC) and Canadian Border Services Agency (CBSA) use Notify for different usecases. The same short-code would be used by both services. This leads to problems with subscriptions, as an end user might chose to unsubscribe from notifications sent by IRCC but inadvertently blocks notifications sent by CBSA.
+3. Short codes are very expensive. As of Sept 2021, a short code is 3000USD$ to setup and a 1000USD$ in recurring monthly costs. It would be cost prohibitive to get short codes per service that uses Notify.
+
+## Potential Solutions
+
+### Unsubscribe Option
+
+The main problem that arises with short codes is that an individual might unsubscibe themselves from all notifications from the Canadian Government, vs the individual service they were interested in unsubscribing from.  
+We could send a shortened link to the user to try for unsubscribing vs a 'STOP' message. The unsubscribe list would be managed through the individual Notify services.
+
+#### Product work required:
+* Need to get a URL shortner that is Gov of Canada owned and used by multiple different departments.
+* Need to build out functionality for managing subscriptions within the Notify Admin.
+
+
+### Multiple Short Codes
+Get multiple short codes - one per service. Currently, there are no Shared Short Codes in Canada. USA allows customers to use 10DLC (https://messagemedia.com/us/blog/shared-short-code-replaced-with-10dlc/) but there isn't an equivalent in Canada. Unless we are willing to use the same short code for every client, we should instead get a short code per client.
+
+
+### Cost of Short Codes
+Short codes are expensive. Short codes require paperwork and 10-16 weeks to get processed. We do not have a self serve option for Notify clients to pick whether they want a Short Code or not. We would have to create a process that would inform individual services about the cost of obtaining a short code, as well as a process for application and applying the Short Code to our service.
+
+
+### Increase Throughput
+Currently we have multiple phone numbers to get over the 1sms/1second rule that is enforced by AWS.
+We do not have another option for increasing throughput unless we apply for the Short Code.
```

Comment:
Ah we mention this here, but that should be in the context I think as this is a goal and not a solution per se.